runtime/gc_structs/ClassArrayClassSlotIterator.cpp

```diff
@@ -50,6 +50,12 @@ GC_ClassArrayClassSlotIterator::nextSlot()
 				_state += 1;
 			}
 			break;
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+		case classArrayClassSlotIterator_state_nullRestrictedArrayClass:
+			classPtr = ((J9ArrayClass *)_iterateClazz)->companionArray;
+			_state += 1;
+			break;
+#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
```

Comment:
Should we have another state to return `nullRestrictedArrayClass` as well?

Reply:
Yes, this is missing the case where a nullRestrictedArrayClass exists and an arrayClass doesn't.

Reply:
I updated this file.

Reply:
Please explain why `nullRestrictedArrayClass` should not be returned? Currently iterator returns all slots of `J9ArrayClass` can contain pointer to another class regardless any application logic.

Reply:
Okay. I will add a case for nullRestrictedArrayClass then since it may contain a pointer.

Reply:
Updated.